### PR TITLE
[EdgeTPU] Add EdgeTPUCompiler

### PR DESCRIPTION
**Pull request by profornnan**
Implemented the EdgeTPUCompiler for EdgeTPUToolchain
Implemented code to test the EdgeTPUCompiler
- constructor
- getToolchainTypes
- parseVersion
- getToolchains
- getInstalledToolchains
- prerequisitesForGetToolchains

ONE-vscode-DCO-1.0-Signed-off-by: profornnan <profornnan@naver.com>

### CODE REVIEW

**Review by rhqjatn2398 — APPROVED** [review]
LGTM 👍

**Comment by dayo09** [comment]
@jyoungyun PTAL :-D

File: src/Backend/EdgeTPU/EdgeTPUToolchain.ts [hunk 1]
```diff
@@ -77,4 +83,191 @@ class EdgeTPUDebianToolchain extends DebianToolchain {
...
+    minor = Number(minor);
+    patch = Number(patch);
+
+    if (isNaN(major) || isNaN(minor) || isNaN(patch)) {
+      throw Error("Invalid version format.");
+    }
+
+    if (splitedVersion.length === 2 && !option) {
+      return new Version(major, minor, undefined);
+    }
```
Comment (dayo09) [inline src/Backend/EdgeTPU/EdgeTPUToolchain.ts:140]: Q. What are these lines for? It seems that patch will be '0' by default when the split length is 2.

**Comment by dayo09** [comment]
There are too many duplicated lines in OneCompiler and EdgeTPUCompiler class. How about extracting some sharing part? Likewise CfgData did.

File: src/Backend/EdgeTPU/EdgeTPUToolchain.ts (hunk 1 above)
Comment (profornnan) [reply src/Backend/EdgeTPU/EdgeTPUToolchain.ts:140]: The latest version of EdgeTPU Compiler is 16.0.

Setting the patch version as the default(16.0.0) causes the installation of edgetpu-compiler using aptitude to fail.

Therefore, I have implemented it to be set as undefined if there is no patch.

However, it seems that if we use undefined in cases where there is no patch version, there may be no need for default value settings.

I will try to modifying it while extracting the common parts of the OneCompiler and EdgeTPUCompiler classes.

**Comment by profornnan** [comment]
OneCompiler and EdgeTPUCompiler are already using the Compiler interface.
Therefore, similar to DebianToolchain, I have extracted the common parts into DebianCompiler.
However, it might be difficult to know what parameters are passed when extending.

So, I would like to confirm whether the current implementation direction is suitable, or if there are any other advisable approaches, such as creating an abstract class for common parts like CfgData and passing settings and classes through functions.

**Review by dayo09 — APPROVED** [review]
LGTM for me :-D

**Comment by dayo09** [comment]
@jyoungyun PTAL :-D

**Review by jyoungyun — APPROVED** [review]
LGTM Thank you ! :)

**Pull request merged**